### ISSUE TITLE
🐛 Source Stripe: wrong default_tax_rates datatype on invoices and subscriptions streams

### ISSUE THREAD

**Issue by kev-datams** [airbytehq/airbyte#32273]
### Connector Name

source-stripe

### Connector Version

4.5.1

### What step the error happened?

During the sync

### Relevant information

Hello,

While running the sync, independently from sync mode, the `invoices` and `subscriptions` streams are in failure due to a wrongly typed `default_tax_rates` field.

The error message is:
```
2023-11-07 16:02:55 source > Marking stream subscriptions as STARTED
2023-11-07 16:02:55 source > Syncing stream: subscriptions 
2023-11-07 16:02:57 source > Failed to transform value {'id': 'txr_1Mdc6iEXaHz6OJ7oxdopxI0G', 'object': 'tax_rate', 'active': True, 'country': 'IT', 'created': 1676910052, 'description': None, 'display_name': 'TVA', 'effective_percentage': 22.0, 'inclusive': False, 'jurisdiction': None, 'livemode': True, 'metadata': {}, 'percentage': 22.0, 'state': None, 'tax_type': None} of type 'object' to '['null', 'array']', key path: '.default_tax_rates.0'
```

As destination has been defined to GCS bucket with AVRO file format, it leads to below error:
```
2023-11-07 16:02:57 destination > ERROR i.a.i.b.FailureTrackingAirbyteMessageConsumer(accept):67 Exception while accepting message tech.allegro.schema.json2avro.converter.AvroConversionException: Failed to convert JSON to Avro: Could not evaluate union, field default_tax_rates is expected to be one of these: NULL, ARRAY. If this is a complex type, check if offending field (path: default_tax_rates) adheres to schema: [{id=txr_1Mdc6iEXaHz6OJ7oxdopxI0G, object=tax_rate, active=true, country=IT, created=1676910052, description=null, display_name=TVA, effective_percentage=22, inclusive=false, jurisdiction=null, livemode=true, metadata={}, percentage=22, state=null, tax_type=null}]
        at tech.allegro.schema.json2avro.converter.JsonGenericRecordReader.read(JsonGenericRecordReader.java:131) ~[converter-1.1.0.jar:?]
        ...
        at io.airbyte.integrations.destination.gcs.GcsDestination.main(GcsDestination.java:46) ~[io.airbyte.airbyte-integrations.connectors-destination-gcs-24.0.2.jar:?]
```
Full error log is available [here](https://github.com/airbytehq/airbyte/files/13288579/d081b27e_377e_4121_8a93_57f31dc43dea_job_18_attempt_1_txt.txt).

In local, I already:
- fixed the [invoices.json](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-stripe/source_stripe/schemas/invoices.json) and [subscriptions.json](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-stripe/source_stripe/schemas/subscriptions.json) configuration files
- build and test a custom docker image
- run the docker image on Airbyte instance with success

Successful output logs:
```
2023-11-07 19:33:10 source > Marking stream subscriptions as STARTED
2023-11-07 19:33:10 source > Syncing stream: subscriptions 
2023-11-07 19:33:12 source > Marking stream subscriptions as RUNNING
2023-11-07 19:33:12 destination > INFO i.a.i.d.r.SerializedBufferingStrategy(lambda$getOrCreateBuffer$0):109 Starting a new buffer for stream subscriptions (current state: 1 MB in 4 buffers)
2023-11-07 19:33:12 destination > WARN i.a.i.d.s.a.JsonToAvroSchemaConverter(getAvroSchema):142 Schema name "subscription_items.items" contains illegal character(s) and is standardized to "subscription_items_items"
2023-11-07 19:33:12 destination > WARN i.a.i.d.s.a.JsonToAvroSchemaConverter(getAvroSchema):142 Schema name "tax_rates.items" contains illegal character(s) and is standardized to "tax_rates_items"
2023-11-07 19:33:12 destination > WARN i.a.i.d.s.a.JsonToAvroSchemaConverter(getAvroSchema):142 Schema name "tiers.items" contains illegal character(s) and is standardized to "tiers_items"
2023-11-07 19:33:12 destination > WARN i.a.i.d.s.a.JsonToAvroSchemaConverter(parseSingleType):260 Array field "data" does not specify the items type. It will default to an array of strings
2023-11-07 19:33:12 destination > WARN i.a.i.d.s.a.JsonToAvroSchemaConverter(getAvroSchema):142 Schema name "default_tax_rates.items" contains illegal character(s) and is standardized to "default_tax_rates_items"
2023-11-07 19:33:17 source > Read 301 records from subscriptions stream
2023-11-07 19:33:17 source > Marking stream subscriptions as STOPPED
2023-11-07 19:33:17 source > Finished syncing subscriptions
```

Thanks ! 👍 

### Contribute

- [X] Yes, I want to contribute
```[tasklist]
### Tasks
- [ ] https://github.com/airbytehq/airbyte/pull/32274
```

**Comment by kev-datams** [comment]
Can somebody add the tag `connectors/source/stripe` please ?

**Comment by kev-datams** [comment]
Fixed via #32286 => use 5.0.0 version
